lib/spack/spack/util/compression.py

```diff
@@ -282,38 +307,108 @@ def decompressor_for(path, extension=None):
 unrecognized file extension: '%s'"
             % extension
         )
+    return extension
 
-    if re.match(r"\.?zip$", extension) or path.endswith(".zip"):
+
+def decompressor_for_nix(extension):
+    """Returns a function pointer to appropriate decompression
+    algorithm based on extension type and unix specific considerations
+    i.e. a reasonable expectation system utils like gzip, bzip2, and xz are
+    available
+
+    Args:
+        path (str): path of the archive file requiring decompression
+    """
+    if re.match(r"zip", extension):
         return _unzip
 
     if re.match(r"gz", extension):
         return _gunzip
 
-    if re.match(r"bz2", extension):
+    if re.match(r"bz", extension):
         return _bunzip2
 
     # Python does not have native support
     # of any kind for .Z files. In these cases,
     # we rely on external tools such as tar,
     # 7z, or uncompressZ
-    if re.match(r"Z$", extension):
+    if re.match(r"Z", extension):
```

Comment:
Yeah not sure why I took this out, but something like `.Z.<whatever other extension` also would not make it to this point in the code, but I'll make the change for conceptual modeling.

Reply:
We should, by this reasoning, always have a `$` at the end of each of the regex's, which is not something that was done prior to Spack on Windows, so I just want to confirm that this is correct.

Reply:
> We should, by this reasoning, always have a $ at the end of each of the regex's, which is not something that was done prior to Spack on Windows, so I just want to confirm that this is correct.

It is. I noticed the issue since this PR removes the trailing `$` in one case, which I assume is unnecessary; I agree with you it should always be fine (and probably clearer) to use `$` in the regexes

Reply:
Sounds good. already done locally.